File: Examples/ClassDiagram/Swift/Song.swift
import Foundation

/// A single track by an artist.
public final class Song: MediaItem {
    public let artist: String
    public let album: String?

    public init(title: String, duration: TimeInterval, genre: Genre, artist: String, album: String? = nil) {
        self.artist = artist
        self.album = album
        super.init(title: title, duration: duration, genre: genre)
    }

    public override func play() {
        print("♪ \(artist) — \(title)")
    }
}

File: Examples/ClassDiagram/Python/song.py
from media_item import MediaItem
from playable import Genre


class Song(MediaItem):
    """A single track by an artist."""

    def __init__(self, title: str, duration: float, genre: Genre, artist: str, album: str | None = None):
        super().__init__(title, duration, genre)
        self.artist: str = artist
        self.album: str | None = album

    def play(self) -> None:
        print(f"♪ {self.artist} — {self.title}")
